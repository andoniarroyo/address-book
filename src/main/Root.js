// @flow
// external types
import type { Store } from 'redux';
// external modules
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
// internal modules
import AddresBook from './Container';
import ContactDetails from './components/ContactDetailsContainer';
import InvalidContact from './components/InvalidContact';

type RootPropsType = {
  store: Store
}

const Root = ({ store }: RootPropsType): React.Element<*> => (
  <Provider store={store}>
    <Router>
      <Switch>
        <Route exact path="/" component={AddresBook} />
        <Route path="/add" component={ContactDetails} />
        <Route path="/edit/:id?" component={ContactDetails} />
        <Route path="/invalid" component={InvalidContact} />
      </Switch>
    </Router>
  </Provider>
);

export default Root;
