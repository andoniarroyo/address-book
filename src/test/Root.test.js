import React from 'react';
import renderer from 'react-test-renderer';

import Root from '../main/Root';

describe('<Root />', () => {
  it('renders the root component', () => {
    const props = {
      store: {
        subscribe: () => {},
        dispatch: () => {},
        getState: () => {},
        setState: () => {},
      },
    };

    const tree = renderer.create(<Root {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
