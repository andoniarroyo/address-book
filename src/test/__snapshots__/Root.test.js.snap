// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> renders the root component 1`] = `null`;
